build/teamcity/cockroach/nightlies/roachtest_compile_bits.sh

```diff
@@ -16,7 +16,7 @@ chmod a+w bin/cockroach bin/roachtest bin/workload
 # Stage the geos libs in the appropriate spot.
 mkdir -p lib.docker_amd64
 chmod o+rwx lib.docker_amd64
-cp $BAZEL_BIN/c-deps/libgeos/lib/libgeos.so   lib.docker_amd64
-cp $BAZEL_BIN/c-deps/libgeos/lib/libgeos_c.so lib.docker_amd64
+cp $BAZEL_BIN/c-deps/libgeos_foreign/lib/libgeos.so   lib.docker_amd64
```

Comment:
I think you need to update the `bazel build` call at the beginning to add the flag `--config force_build_cdeps`.